### PR TITLE
✨ Add ISCN tx Fee Calculation

### PR DESCRIPTION
**Pull request by shellyYG**

### CODE REVIEW

File: locales/client/en.json [hunk 1]
```diff
@@ -1035,7 +1035,8 @@
       "title": "Title",
       "type": "Content Type",
       "license": "License",
-      "tags": "Tags"
+      "tags": "Tags",
+      "ISCNTotalFee": "Total ISCN Gas Fee: {ISCNTotalFee} LIKE"
```
Comment (williamchong) [inline locales/client/en.json]:
```suggestion
      "ISCNTotalFee": "Estimated ISCN Fee: {ISCNTotalFee} LIKE"
```

File: store/modules/actions/payment.js [hunk 2]
```diff
@@ -61,6 +64,38 @@ export async function sendCosmosPayment(
   }
 }
 
+export async function calISCNTxTotalFee({ commit },
```
Comment (williamchong) [inline store/modules/actions/payment.js:67]:
```suggestion
export async function calculateISCNTxTotalFee({ commit },
```

File: util/cosmos/iscn/sign.js [hunk 3]
```diff
@@ -125,6 +124,79 @@ function formatISCNPayload(payload, version = 1) {
...
+  const byteSize = Buffer.from(jsonStringify(obj), 'utf-8').length
+    + stakeholders.reduce((acc, s) => acc + s.length, 0)
+    + contentMetadata.length;
+  const feeAmount = byteSize * feePerByteAmount;
+
+  return feeAmount;
+}
+
+export async function estimateISCNTxGas(tx, version) {
+  const DEFAULT_GAS = 1000000; // TODO: estimate according to size
```
Comment (williamchong) [inline util/cosmos/iscn/sign.js:182]: Please refactor this to `ISCN_GAS` to `@/constant`, also would `200000` be enough?

File: util/cosmos/iscn/sign.js [hunk 4]
```diff
@@ -125,6 +124,79 @@ function formatISCNPayload(payload, version = 1) {
...
+  return {
+    gasFee: {
+      amount: [{ amount: (DEFAULT_GAS_PRICE_NUMBER * DEFAULT_GAS).toFixed(), denom: 'nanolike' }],
+      gas: DEFAULT_GAS.toFixed(),
+    },
+    iscnFee: ISCN_FEE,
+  };
+}
+
+export async function calISCNTotalFee(tx) {
```
Comment (williamchong) [inline util/cosmos/iscn/sign.js:193]:
```suggestion
export async function calculateISCNTotalFee(tx) {
```

**Review by williamchong — CHANGES_REQUESTED** [review]

File: util/cosmos/iscn/sign.js (hunk 3 above)
Comment (shellyYG) [reply util/cosmos/iscn/sign.js:182]: I think so. So far my test tx gas_used are all < 150,000 nanolike

File: pages/in/widget/iscn/index.vue [hunk 5]
```diff
@@ -287,6 +295,33 @@ export default {
     },
   },
   async mounted() {
+    const showDialogAction = !this.redirectUri;
```
Comment (williamchong) [inline pages/in/widget/iscn/index.vue:298]: What is this `showDialogAction` for? It is not used in `calculateISCNTxTotalFee` seems

**Review by williamchong — CHANGES_REQUESTED** [review]

File: util/cosmos/iscn/sign.js [hunk 6]
```diff
@@ -125,6 +125,78 @@ function formatISCNPayload(payload, version = 1) {
...
+  const ISCN_FEE = await estimateISCNTxFee(tx, { version });
+  return {
+    gasFee: {
+      amount: [{ amount: (DEFAULT_GAS_PRICE_NUMBER * ISCN_GAS).toFixed(), denom: 'nanolike' }],
+      gas: ISCN_GAS.toFixed(),
+    },
+    iscnFee: ISCN_FEE,
+  };
+}
+
```
Comment (williamchong) [inline util/cosmos/iscn/sign.js:192]: Would separating `estimateISCNTxFee()` and `estimateISCNTxGas()` be clearer? 
Remove `iscnFee` from `estimateISCNTxGas`, and just call two functions if we want total
```
const gasFee = await estimateISCNTxGas(tx);
const iscnFee = await estimateISCNTxFee();
```
when using

**Review by williamchong — CHANGES_REQUESTED** [review]

File: util/cosmos/iscn/sign.js (hunk 6 above)
Comment (shellyYG) [reply util/cosmos/iscn/sign.js]: Yes, good idea :)

**Review by williamchong — APPROVED** [review]

**Pull request merged**